### PR TITLE
[JENKINS-31691] isUnix step

### PR DESCRIPTION
**Pull request by jglick**
[JENKINS-31691](https://issues.jenkins-ci.org/browse/JENKINS-31691)

@reviewbybees esp. @jtnord

### CODE REVIEW

File: basic-steps/src/main/resources/org/jenkinsci/plugins/workflow/steps/IsUnixStep/help.html [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<div>
+    Returns true if enclosing <code>node</code> is running on a Unix system, false if Windows.
```
Comment (jtnord) [inline basic-steps/src/main/resources/org/jenkinsci/plugins/workflow/steps/IsUnixStep/help.html]: Not. Unix like system. Linux is not Unix

File: basic-steps/src/main/java/org/jenkinsci/plugins/workflow/steps/IsUnixStep.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        public DescriptorImpl() {
+            super(Execution.class);
+        }
+
+        @Override public String getFunctionName() {
+            return "isUnix";
+        }
+
+        @Override public String getDisplayName() {
+            return "Checks if running on a Unix node";
```
Comment (jtnord) [inline basic-steps/src/main/java/org/jenkinsci/plugins/workflow/steps/IsUnixStep.java]: Unix like?

**Comment by jtnord** [comment]
Code looks good so :bee: there.
Nits that Linux is not Unix and the help/docs says Unix not Unix -like.

**Comment by amuniz** [comment]
:bee:

**Comment by jtnord** [comment]
:bee:

**Pull request merged**